### ISSUE TITLE
Regex fails for kernel-version.full on Raspberry Pi nodes

### ISSUE THREAD

**Issue by dbaker-rh** [kubernetes-sigs/node-feature-discovery#390]
**What happened**:

Error in logs:
2020/11/18 23:39:01 Ignoring invalid feature value kernel-version.full=5.4.51-v8_: [a valid label must be an empty string or consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyValue',  or 'my_value',  or '12345', regex used for validation is '(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?')]

Additionally, feature.node.kubernetes.io/kernel-version.full does not get applied to the node

**What you expected to happen**:

Kernel version should be correctly recognized.

**How to reproduce it (as minimally and precisely as possible)**:

My reproduction environment:
- install k3os (k3s mini kubernetes distro) on raspberry pi 4
- install nfd (note: since this is arm64 I cross-compiled the container)
- watch logs

On the host:

\# uname -r
5.4.51-v8+

Specifically, the + on the end of the kernel string gets turned into an _ here
- https://github.com/kubernetes-sigs/node-feature-discovery/blob/master/source/kernel/kernel.go#L115
- fullRegex := regexp.MustCompile("[^-A-Za-z0-9_.]")

Then, the resulting string with a trailing underscore does not meet the required syntax for a feature value.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): 

Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.9+k3s1", GitCommit:"630bebf94b9dce6b8cd3d402644ed023b3af8f90", GitTreeState:"clean", BuildDate:"2020-09-17T19:05:03Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/arm64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.9+k3s1", GitCommit:"630bebf94b9dce6b8cd3d402644ed023b3af8f90", GitTreeState:"clean", BuildDate:"2020-09-17T19:05:03Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/arm64"}

- Cloud provider or hardware configuration: raspberry pi 4
- OS (e.g: `cat /etc/os-release`):  VERSION="k3OS v0.11.1"
- Kernel (e.g. `uname -a`): Linux pi4-master 5.4.51-v8+ #1333 SMP PREEMPT Mon Aug 10 16:58:35 BST 2020 aarch64 GNU/Linux

- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

**Comment by marquiz** [comment]
I see 🤔  Maybe the correct fix would be to just drop the invalid characters from the end of the value. I.e. in this case the label would be `kernel-version.full=5.4.51-v8`. Another solution would be to preped some alnum like `0` and it would become `kernel-version.full=5.4.51-v8_0` but somehow that looks even uglier. WDYT?

**Comment by dbaker-rh** [comment]
Where does the requirement come from for the value to not *end* with non-alphanumeric.

It seems we can't do anything that makes sense here.  If we have to edit the value to fit the regex there should be some way to indicate it's edited, and that doesn't work if we can't use a special character to indicate a removed or edited character.

For whatever it's worth, in this case the full kernel string is stored elsewhere ...
```
  kubectl get nodes -o json | jq '.items[].status.nodeInfo.kernelVersion'
  "5.4.51-v8+"
```

As the least bad option, I think making sure the first regex that sanitizes the string should at least put it in a form that can be used later.  If that means dropping the characters entirely then, perhaps, that's least bad than both making a change and also generating an error later.

**Comment by marquiz** [comment]
> Where does the requirement come from for the value to not _end_ with non-alphanumeric.

The limitation comes from the Kubernetes API. Label values must start and end with an alphanumeric.

> As the least bad option, I think making sure the first regex that sanitizes the string should at least put it in a form that can be used later.

Yeah, sure. It's a clear bug in NFD that it sanitizes the value into an invalid form

> If that means dropping the characters entirely then, perhaps, that's least bad than both making a change and also generating an error later.

K, I'll make a change trimming invalid characters

**Comment by marquiz** [comment]
@dbaker-rh should be fixed by #402
